enterprise_gateway/services/processproxies/container.py

```diff
@@ -230,8 +237,13 @@ def get_initial_states(self):
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_container_status(self, iteration_string):
-        """Return current container state."""
+    def get_error_states(self):
+        """Return list of states indicating container is starting (includes running)."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_container_status(self, iteration: int | None) -> str:
+        """Return current container state or empty string."""
```

Comment:
Let's emphasize the contract...
```suggestion
        """Returns the current container state (in lowercase) or the empty string if not available."""
```